twoliter/src/lock.rs

```diff
@@ -139,6 +170,111 @@ impl Lock {
+        let mut kits: Vec<Image> = Vec::new();
+        for image in self.kit.iter() {
+            kits.push(self.extract_kit(&target_dir, image, arch).await?);
+        }
+        let kit_list = serde_json::to_vec(&ExternalKitMetadata {
+            sdk: Image {
+                name: self.sdk.name.clone(),
+                version: self.sdk.version.clone(),
+                vendor: self.sdk.vendor.clone(),
+            },
+            kits,
+        })
+        .context("failed to serialize external kit metadata")?;
+        write(
+            target_dir.join("external-kit-metadata.json"),
+            kit_list.as_slice(),
+        )
+        .await
+        .context("failed to write external kit metadata")?;
+
+        Ok(())
+    }
+
+    async fn extract_kit<P>(&self, path: P, image: &LockedImage, arch: &str) -> Result<Image>
+    where
+        P: AsRef<Path>,
+    {
+        let tmp_dir =
+            tempdir().context("failed to create a temporary directory for kit extraction")?;
+        let vendor = image.vendor.clone();
+        let name = image.name.clone();
+        let docker_arch = match arch {
+            "x86_64" => Ok("amd64"),
+            "aarch64" => Ok("arm64"),
+            _ => Err(anyhow!("unsupported architecture '{}'", arch)),
+        }?;
+        let target_path = path.as_ref().join(format!("{vendor}/{name}/{arch}"));
+        create_dir_all(&target_path).await?;
+
+        // First we want to get the correct digest for our architecture
+        let manifest_bytes = docker!(
+            ["manifest", "inspect", image.source.as_str()],
+            format!("failed to find a kit {}", image.to_string())
+        );
+        let manifest_list: ManifestListView = serde_json::from_slice(manifest_bytes.as_slice())
+            .context("failed to deserialize manifest list")?;
+        let manifest = manifest_list
+            .manifests
+            .iter()
+            .find(|x| x.platform.as_ref().unwrap().architecture == docker_arch)
+            .context(format!(
+                "could not find kit image for architecture '{}' at {}",
+                arch, image.source
+            ))?;
+        let image_source = image.source.replace(
+            format!(":v{}", image.version).as_str(),
+            format!("@{}", manifest.digest).as_str(),
+        );
+        let oci_archive = docker!(
+            ["save", image_source.as_str(),],
+            format!("failed to fetch kit and save to disk from {}", image_source)
+        );
+        let mut oci_archive = Archive::new(oci_archive.as_slice());
+        // Unpack the oci image to tempdirectory
+        oci_archive
+            .unpack(tmp_dir.path())
+            .context("failed to unpack oci image")?;
+        // Now read in the index
+        let index_bytes = read(tmp_dir.path().join("index.json")).await?;
+        let index: IndexView = serde_json::from_slice(index_bytes.as_slice())
+            .context("failed to deserialize oci image index")?;
```

Comment:
This looks like a function.

```suggestion
        let index = self.get_index(x, y, z)?;
```